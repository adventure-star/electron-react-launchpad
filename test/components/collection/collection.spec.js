import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Collection from '../../../src/renderer/components/collection/collection'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<Collection className='testClass'>Test collection content</Collection>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Collection component', () => {
    it('should be of type ul', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('ul');
    });

    it('should have only the classes "collection" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('collection testClass');
    });

    it('children should be the text "Test card content"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test collection content');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
