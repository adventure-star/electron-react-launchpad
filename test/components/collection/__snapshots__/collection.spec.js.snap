// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection component Matches/Renders a snapshot 1`] = `
<ul
  className="collection testClass"
>
  Test collection content
</ul>
`;
